src/components/Header.js

```diff
@@ -47,7 +43,8 @@ function Header({handleButtonClick}) {
         {!isLoggedIn && (
           <button className="login-button" onClick={() => setLoginOpen(true)}>Log In</button>
           )}
-        {isLoggedIn && <p>Welcome to AI AT Ally, {currentUser.name}! You have successfully logged in!</p>}
+        {isLoggedIn && currentUser && <p>Welcome, {currentUser.name}! You have successfully logged in!</p>}
```

Comment:
This is really nice, Nicole. I like how personalized this will be for the end user.